File: my-folder/0733-flood-fill/solution.java
import java.util.HashSet;

class Solution {
    HashSet<Integer[]> filled = new HashSet<>();
    
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int pix_color = image[sr][sc];
        image[sr][sc] = color + 1;
        int[][] directions = new int[][] {
            {0, 1}, {0, -1}, {1, 0}, {-1, 0}};
        for (int[] direction : directions) {
            int new_sr = sr + direction[1];
            int new_sc = sc + direction[0];
            if (new_sc >= image[0].length || new_sc < 0) {
                continue;
            }
            if (new_sr >= image.length || new_sr < 0) {
                continue;
            }
            if (image[new_sr][new_sc] == pix_color) {
                floodFill(image, new_sr, new_sc, color);
            }
        }
        image[sr][sc] = color;
        return image;
    }
}

File: my-folder/0733-flood-fill/solution.py
class Solution(object):
    def floodFill(self, image, sr, sc, color):
        """
        :type image: List[List[int]]
        :type sr: int
        :type sc: int
        :type color: int
        :rtype: List[List[int]]
        """
        filled = set()
        to_fill = [[sr, sc]]
        moves = [[-1, 0], [1, 0], [0, -1], [0, 1]]
        original_color = image[sr][sc]

        while len(to_fill) > 0:
            cell = to_fill.pop()
            row = cell[0]
            col = cell[1]
            if (image[row][col] != original_color):
                continue
            if (row, col) in filled:
                continue
            else:
                filled.add((row, col))
            image[row][col] = color
            print("ADD")
            for move in moves:
                adjacent_tile = [row + move[0], col + move[1]]
                if self.isValidTile(image, adjacent_tile):
                    to_fill.append(adjacent_tile)
        return image

            

    def isValidTile(self, image, cell):
        row = cell[0]
        col = cell[1]
        rows = len(image)
        cols = len(image[0])
        return 0 <= row < rows and 0 <= col < cols

